src/ray/raylet/scheduling/cluster_task_manager.cc

```diff
@@ -275,21 +279,90 @@ bool ClusterTaskManager::CancelTask(const TaskID &task_id) {
 void ClusterTaskManager::FillResourceUsage(
     bool light_report_resource_usage_enabled,
     std::shared_ptr<rpc::ResourcesData> data) const {
+  if (max_resource_shapes_per_load_report_ == 0) {
+                                                  return;
+  }
   // TODO (WangTao): Find a way to check if load changed and combine it with light
   // heartbeat. Now we just report it every time.
   data->set_resource_load_changed(true);
   auto resource_loads = data->mutable_resource_load();
   auto resource_load_by_shape =
       data->mutable_resource_load_by_shape()->mutable_resource_demands();
 
-  // TODO (Alex): Implement the 1-CPU task optimization.
+  int num_reported = 0;
+
+  // 1-CPU optimization
+  static const ResourceSet one_cpu_resource_set(
+      std::unordered_map<std::string, double>({{kCPU_ResourceLabel, 1}}));
+  static const SchedulingClass one_cpu_scheduling_cls(
+      TaskSpecification::GetSchedulingClass(one_cpu_resource_set));
+  {
+    num_reported++;
+    int count = 0;
+    int backlog_size = 0;
+    auto it = tasks_to_schedule_.find(one_cpu_scheduling_cls);
+    if (it != tasks_to_schedule_.end()) {
+      count += it->second.size();
+      const auto &work_queue = it->second;
+      for (const auto &work : work_queue) {
+        const auto &task = std::get<0>(work);
+        backlog_size += task.BacklogSize();
```

Comment:
It's not correct to be adding up the backlog sizes right? Or is this assuming core worker only sends one task per type?

If the latter, we should add a comment about the assumptions we are making here.

Reply:
Yeah we're making that assumption (same assumption was made in the old scheduler). 

Avoiding that assumption would be very hard/annoying actually, since the raylet would have to track the backlog per core worker per shape (and we would have to de-dupe backlogs from the same core worker to multiple raylets).